packages/dds/tree/src/test/utils.ts

```diff
@@ -111,37 +151,34 @@ export class TestTreeProvider {
 
     /**
      * Give this {@link TestTreeProvider} the ability to summarize on demand during a test by creating a summarizer
-     * client for the container at the given index. This must be called before any trees submit any edits, or else a
-     * different summarizer client might already have been elected.
-     * @param index - the container that will spawn the summarizer client
-     * @returns a function which will cause a summary to happen when awaited. May be called multiple times.
+     * client for the container at the given index.  This can only be called when the summarizeOnDemand parameter
+     * was set to true when calling the create() method.
+     * @returns void after a summary has been resolved. May be called multiple times.
      */
-    public async enableManualSummarization(index = 0): Promise<() => Promise<void>> {
-        assert(index < this.trees.length, "Index out of bounds: not enough trees");
-        const summarizer = await createSummarizer(this.provider, this.containers[index]);
-        return async () => {
-            await summarizeNow(summarizer, "TestTreeProvider");
-        };
+    public async summarize(): Promise<void> {
+        assert(
+            this.summarizer !== undefined,
+            "can't summarize because summarizeOnDemand was not set to true.",
+        );
+        await summarizeNow(this.summarizer, "TestTreeProvider");
     }
 
     public [Symbol.iterator](): IterableIterator<ISharedTree> {
         return this.trees[Symbol.iterator]();
     }
-
-    private constructor() {
-        const factory = new SharedTreeFactory();
-        const registry = [[TestTreeProvider.treeId, factory]] as ChannelFactoryRegistry;
-        const driver = new LocalServerTestDriver();
-        this.provider = new TestObjectProvider(
-            Loader,
-            driver,
-            () =>
-                new TestContainerRuntimeFactory(
-                    "@fluid-example/test-dataStore",
-                    new TestFluidObjectFactory(registry),
-                ),
-        );
-
+    //
+
+    private constructor(
+        provider: ITestObjectProvider,
+        firstTreeParams?: [IContainer, ISharedTree, ISummarizer],
+    ) {
+        this.provider = provider;
+        if (firstTreeParams !== undefined) {
+            const [container, firstTree, summarizer] = firstTreeParams;
+            this._containers.push(container);
+            this._trees[this.trees.length] = firstTree;
```

Comment:
use `push` here. It's written this funny way on line 147 because it's being used as a return value there.

Reply:
Updated to use push